freqtrade/optimize/backtesting.py

```diff
@@ -591,6 +592,42 @@ def _get_sell_trade_entry(self, trade: LocalTrade, sell_row: Tuple) -> Optional[
         else:
             return self._get_sell_trade_entry_for_candle(trade, sell_row)
 
+    def _leverage_prep(
```

Comment:
is this method different to the one in freqtradebot, or just a plain copy?
maybe we have a possibility to extract this to a neutral location if it's identical? (it looks identical at first glance ...)

Reply:
It's identical

Reply:
We could separate it if we pass `exchang` `trading_mode` and `margin_mode` as a parameter

Reply:
we can also just move it to the exchange class directly :laughing:  it got everything we need and more.